src/http.rs

```diff
@@ -240,18 +241,18 @@ pub enum Auth {
     /// The username and password are concatenated and encoded via base64.
     Basic {
         /// The username to send.
-        user: String,
+        user: SensitiveString,
```

Comment:
This could be left as a string, I'd say?

If it wouldn't normally be hidden on a webpage/HTML form, it probably shouldn't be hidden in Vector.